### DISCUSSION TITLE
provide "hazard sides" in the json payload

### DISCUSSION THREAD

**Discussion by fables-tales** [BattlesnakeOfficial/feedback#159] [Feature Requests]
Right now, in the JSON payload for move requests, hazards are represented as an array of cells. This makes determining how far in the hazards have spawned from each side is difficult. The reason why I need this is that in order to simulate hazards coming in from the side more easily, I'd like to be able to easily determine the state on the battlesnake engine side, as opposed I suggest, along with the actual locations of hazards, providing a `hazard_sides` object which is like:

```
"board": {
    ...
    "hazards": [
      ...
    ],
    "hazard_sides": {
        "right": 0,
        "left": 4,
        "top": 1,
        "bottom": 0,
    }
```
this would tell us how far in from each side hazards have been added, representing something like this:
```
x x x x x x x x x x x
x x x x . . . . . . . 
x x x x . . . . . . . 
x x x x . . . . . . . 
x x x x . . . . . . . 
x x x x . . . . . . . 
x x x x . . . . . . . 
x x x x . . . . . . . 
x x x x . . . . . . . 
x x x x . . . . . . . 
x x x x . . . . . . . 
```

**Comment by bvanvugt** [comment]
I'm all for this existing 👍  but it'll be a low priority on our end given that it can be calculated relatively easy/cheaply.

For anyone interested in how this is tracked in the game engine, the relevant code is available here:
https://github.com/BattlesnakeOfficial/rules/blob/main/royale.go#L37
